### PR TITLE
[Feature]: First Phase of Expense Tracker Application

### PR DESCRIPTION
**Pull request by supminn**
## What does this PR do?

First phase of Expense tracker application which has the following features
- User authentication using [nhost](https://nhost.io/).
- Income and expense data stored and accessible to authenticated users.
- Landing page, welcome page, login, and registration pages.

### CODE REVIEW

**Comment by DanishOnGit** [comment]
Hey @supminn , overall functionality looks fine.🚀 Just some minor observations:
**Some bugs**:

- I am able to enter negative numbers in the income/expense form
- A bug in showing the balance amount  (maybe result of above bug)
- When list items in screen increase, then the list item card hides behind the add entity Plus button
- While refreshing after logging in, there is a flash of Login screen

**Some design suggestions**:

- The Income and Expense lists might be better in a 2 column view. Why? Say, in Daily view, I have 30 items on Income side on 15Oct,2022. Now, to see the the Expense list for the same day, I have to scroll a lot to the bottom.
This can be solved by a 2 column view(Income and Expense). Wdyt?
- The daily, monthly, yearly buttons can come close together as a button group on the right side of the screen, so that user does not have to move the cursor across the entire screen width to click the buttons
- The left and right arrow btns can also come closer to reduce cursor movement efforts
- In month view, the cards can be arranged in ascending date order, wdyt?
- minor suggestion, I think the Profile icon can be changed to a door like button, so that it clearly looks like a logout button

**Comment by supminn** [comment]
Thanks a lot, @DanishOnGit for taking out your time and providing a detailed review.
I have incorporated the fixes and considered your suggestions 😄
Do let me know if you find any more observations!

File: README.md [hunk 1]
```diff
@@ -9,10 +9,9 @@ This project would be a part of [Hack-R-Play](https://hustles.reactplay.io/hackr
 
 ## Tech stack
 
-- Next.JS
-- nHost (backend)
-- tailwind?
--
+- Next.JS for user-facing application
+- nHost for data storage and authentication
+- TailwindCSS for UI styling
```
Comment (supminn) [inline README.md:40]: Note to self
- Add feature-level details
- Mention the upcoming features
- Steps to raise an issue to report a bug or refer to CONTRIBUTING.md

File: components/Modals/ComingSoon.jsx [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+              </p>
+            </div>
+            <div className="bg-gray-50 px-4 py-3 sm:flex sm:flex-row-reverse sm:px-6">
+              <button
+                onClick={() => setShowModal(false)}
+                type="button"
+                className="mt-3 inline-flex w-full justify-center rounded-md border border-gray-300 bg-white px-4 py-2 text-base font-medium text-gray-700 shadow-sm hover:bg-gray-50 sm:mt-0 sm:ml-3 sm:w-auto sm:text-sm"
+              >
+                Close
+              </button>
```
Comment (supminn) [inline components/Modals/ComingSoon.jsx:35]: Suggestion: Create a common `Modal` component that would take props about `title` and `children` components.

File: components/Welcome.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+          "linear-gradient(rgba(255, 255, 255, 0.6), rgba(255, 255, 255, 0.4)), url(/background_img.jpg)",
+      }}
+    >
+      <section className="text-center">
+        <h1 className="text-4xl text-green-900 italic">Welcome</h1>
+        <h1 className="text-4xl text-green-900 italic">to</h1>
+        <div className="m-3">
+          <Image priority src="/logo.png" alt="" height={300} width={550} />
+          <p className="p-4 text-xl text-green-900 font-bold">
+            One stop shop to track your income and expenses
```
Comment (supminn) [inline components/Welcome.jsx]: Add more details describing the features of this application.

File: components/YearlyReport.jsx [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+          </tbody>
+        </table>
+      </section>
+    </>
+  );
+};
+
+const MonthlySummary = ({ month, setRoute }) => {
+  const {
+    dataState: { selectedDate, incomeData: income, expenseData: expense },
```
Comment (supminn) [inline components/YearlyReport.jsx]: Renaming income and expense is not required in this file

File: components/index.js [hunk 5]
```diff
@@ -0,0 +1,7 @@
+export { Welcome } from "./Welcome";
+export { Landing } from "./Landing";
+export { Header } from "./Header";
+export { Footer } from "./Footer";
+export { PrimaryButton } from "./Buttons";
+export { SecondaryButton } from "./Buttons";
+export { Password } from "./Password";
```
Comment (supminn) [inline components/index.js:7]: Add DailyReport, MonthlyReport, and YearlyReport to this file.

File: pages/api/hello.js [hunk 6]
```diff
@@ -1,5 +1,5 @@
 // Next.js API route support: https://nextjs.org/docs/api-routes/introduction
 
 export default function handler(req, res) {
-  res.status(200).json({ name: 'John Doe' })
+  res.status(200).json({ name: "John Doe" });
 }
```
Comment (supminn) [inline pages/api/hello.js]: This file can be deleted.

**Pull request merged**